packages/edit-site/src/hooks/commands/use-edit-mode-commands.js

```diff
@@ -158,7 +168,70 @@ function useManipulateDocumentCommands() {
 	};
 }
 
+function useAddNewPageCommand() {
+	const history = useHistory();
+	const { saveEntityRecord } = useDispatch( coreStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	const createPageEntity = async ( { close } ) => {
+		try {
+			const _page = await saveEntityRecord(
+				'postType',
+				'page',
+				{
+					status: 'draft',
+				},
+				{
+					throwOnError: true,
+				}
+			);
+			if ( _page?.id ) {
+				history.push( {
+					postId: _page.id,
+					postType: 'page',
+					canvas: 'edit',
+				} );
+			}
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while creating the item.' );
+
+			createErrorNotice( errorMessage, {
+				type: 'snackbar',
+			} );
+		} finally {
+			close();
+		}
+	};
+
+	const commands = useMemo( () => {
+		return [
+			{
+				name: 'core/add-new-page',
+				label: __( 'Add new page' ),
+				icon: plus,
+				callback: createPageEntity,
+			},
+		];
+	}, [ createPageEntity ] );
+
+	return {
+		isLoading: false,
+		commands,
+	};
+}
+
 export function useEditModeCommands() {
+	const { unregisterCommand } = useDispatch( commandsStore );
+	unregisterCommand( 'core/add-new-page' );
+
+	useCommandLoader( {
+		name: 'core/add-new-page',
+		hook: useAddNewPageCommand,
+	} );
```

Comment:
Instead of doing this "unregister register" hack here. Can we move the check the core-commands component instead? I believe we already have some commands there that check for site editor to do something slightly different.

Reply:
Thanks for the feedback. I had it in the first one or two commits of this PR so it'll be an easy revert. 👍🏻

Reply:
Sorry for the back and forth, Ramon!

Reply:
All good! I appreciate the reviews. Best to explore all the options I reckon. 👍🏻

Reply:
Ah sorry I missed the beginning of the PR. For me both approaches can work, I just like when we're consistent. All commands following the same pattern.

Reply:
Thanks for clarifying, Riad!